From f9aafdacbd273da887ee36c973c967e75bbc38fb Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 7 Aug 2023 11:04:35 +0200
Subject: [PATCH 08/15] wintrust: Use a more capable cryptographic service
 provider in CryptCATAdminEnumCatalogFromHash.

The default provider doesn't even support SHA256 signatures.

Signed-off-by: Torge Matthies <openglfreak@googlemail.com>
---
 dlls/wintrust/crypt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/wintrust/crypt.c b/dlls/wintrust/crypt.c
index 11111111111..11111111111 100644
--- a/dlls/wintrust/crypt.c
+++ b/dlls/wintrust/crypt.c
@@ -301,7 +301,7 @@ HCATINFO WINAPI CryptCATAdminEnumCatalogFromHash(HCATADMIN hCatAdmin, BYTE* pbHa
     }
     if (phPrevCatInfo) prev = *phPrevCatInfo;
 
-    ret = CryptAcquireContextW(&prov, NULL, MS_DEF_PROV_W, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT);
+    ret = CryptAcquireContextW(&prov, NULL, MS_ENH_RSA_AES_PROV_W, PROV_RSA_AES, CRYPT_VERIFYCONTEXT);
     if (!ret) return NULL;
 
     if (!prev)
-- 
0.0.0

